config/kube.py

```diff
@@ -0,0 +1,359 @@
+#!/usr/bin/python -u
+# -*- coding: utf-8 -*-
+
+import click
+import os
+import sys
+import syslog
+import urllib3
+import tempfile
+import requests
+import fcntl
+import sonic_device_util
+import yaml
+import netaddr
+import inspect
+import shutil
+from urlparse import urlparse
+from swsssdk import ConfigDBConnector
+
+def hostname():
+    return os.uname()[1]
+
+from config.main import run_command, AbbreviationGroup, config
+
+logger_identity = "kube_config"
+log_level = syslog.LOG_DEBUG
+
+KUBE_ADMIN_CONF = "/etc/sonic/kube_admin.conf"
+KUBELET_YAML = "/var/lib/kubelet/config.yaml"
+KUBELET_SERVICE = "/etc/systemd/system/multi-user.target.wants/kubelet.service"
+
+SERVER_ADMIN_URL = "https://{}/admin.conf"
+KUBEADM_JOIN_CMD = "kubeadm join --discovery-file /etc/sonic/kube_admin.conf --node-name {}"
+
+LOCK_FILE = "/var/lock/kube_join.lock"
+
+def _log_msg(lvl, fname, ln, m):
+    if lvl <= log_level:
+        syslog.openlog(logger_identity)
+        syslog.syslog(lvl, m)
+        syslog.closelog()
+
+    if (
+            log_level == syslog.LOG_DEBUG or
+            sys.flags.interactive or
+            sys.flags.debug or
+            sys.flags.verbose):
+        print("{}: {}:{}: {}".format(logger_identity, fname, ln, m))
+
+def _log_info(m):
+    _log_msg(syslog.LOG_INFO, inspect.stack()[1][1], inspect.stack()[1][2], m)
+
+def _log_err(m):
+    _log_msg(syslog.LOG_ERR, inspect.stack()[1][1], inspect.stack()[1][2], m)
+
+def _log_debug(m):
+    _log_msg(syslog.LOG_DEBUG, inspect.stack()[1][1], inspect.stack()[1][2], m)
+
+def _do_exit(m):
+    m = "FATAL failure: {}. Exiting...".format(m)
+    _log_msg(syslog.LOG_ERR, inspect.stack()[1][1], inspect.stack()[1][2], m)
+    raise SystemExit(m) 
+
+def _get_db_data(table, key):
```

Comment:
>_get_db_data [](start = 4, length = 12)

Same as `_get_configdb_data()`, reuse?

Reply:
I will defer this to re-factoring, as it is not right to include the file from show into config.

Reply:
You can see, every .py has its own get/set of config DB else duplicated all over. A re-factoring work is required to cleanup all.